scala-package/examples/src/main/java/org/apache/mxnetexamples/infer/javapi/objectdetector/SSDClassifierExample.java

```diff
@@ -0,0 +1,190 @@
+	
+	final static Logger logger = LoggerFactory.getLogger(SSDClassifierExample.class);
+	
+	static List<List<Pair<String, List<Float>>>> runObjectDetectionSingle(String modelPathPrefix, String inputImagePath, List<Context> context) {
+		Shape inputShape = new Shape(new int[] {1, 3, 512, 512});
+		List<DataDesc> inputDescriptors = new ArrayList<DataDesc>();
+		inputDescriptors.add(new DataDesc("data", inputShape, DType.Float32(), "NCHW"));
+		BufferedImage img = ObjectDetector.loadImageFromFile(inputImagePath);
+		ObjectDetector objDetector = new ObjectDetector(modelPathPrefix, inputDescriptors, context, 0);
+		return objDetector.imageObjectDetect(img, 3);
+	}
+	
+	static List<List<List<Pair<String, List<Float>>>>> runObjectDetectionBatch(String modelPathPrefix, String inputImageDir, List<Context> context) {
+		Shape inputShape = new Shape(new int[]{1, 3, 512, 512});
+		List<DataDesc> inputDescriptors = new ArrayList<DataDesc>();
+		inputDescriptors.add(new DataDesc("data", inputShape, DType.Float32(), "NCHW"));
+		ObjectDetector objDetector = new ObjectDetector(modelPathPrefix, inputDescriptors, context, 0);
+		
+		// Loading batch of images from the directory path
+		List<List<String>> batchFiles = generateBatches(inputImageDir, 20);
+		List<List<List<Pair<String, List<Float>>>>> outputList = new ArrayList<List<List<Pair<String, List<Float>>>>>();
+		
+		for (List<String> batchFile : batchFiles) {
+			List<BufferedImage> imgList = ObjectDetector.loadInputBatch(batchFile);
+			// Running inference on batch of images loaded in previous step
+			List<List<Pair<String, List<Float>>>> tmp = objDetector.imageBatchObjectDetect(imgList, 5);
+			outputList.add(tmp);
+		}
+		return outputList;
+	}
+	
+	static List<List<String>> generateBatches(String inputImageDirPath, int batchSize) {
+		File dir = new File(inputImageDirPath);
+
+		List<List<String>> output = new ArrayList<List<String>>();
+		List<String> batch = new ArrayList<String>();
+		for (File imgFile : dir.listFiles()) {
+			batch.add(imgFile.getPath());
+			if (batch.size() == batchSize) {
+				output.add(batch);
+				batch = new ArrayList<String>();
+			}
+		}
+		if (batch.size() > 0) {
+			output.add(batch);
+		}
+		return output;
+	}
+	
+	public static void main(String[] args) {
+		SSDClassifierExample inst = new SSDClassifierExample();
+		CmdLineParser parser  = new CmdLineParser(inst);
+		try {
+			parser.parseArgument(args);
+		} catch (Exception e) {
+			logger.error(e.getMessage(), e);
+			parser.printUsage(System.err);
+			System.exit(1);
+		}
+		
+		String mdprefixDir = inst.modelPathPrefix;
+		String imgPath = inst.inputImagePath;
+		String imgDir = inst.inputImageDir;
+		
+		if (!checkExist(Arrays.asList(mdprefixDir + "-symbol.json", imgDir, imgPath))) {
+			logger.error("Model or input image path does not exist");
+			System.exit(1);
+		}
+		
+		List<Context> context = new ArrayList<Context>();
+		if (System.getenv().containsKey("SCALA_TEST_ON_GPU") &&
```

Comment:
Should this flag still be SCALA_TEST_ON_GPU ?

Reply:
oh... that's a good question

Reply:
okay, I think that this is correct given that the plan is for the java tests to run as part of the scala test suite instead of as a separate thing. This is far less important for this example than others since there isn't any training involved and the whole thing runs in a matter of seconds.